lib/google/protobuf.ex

```diff
@@ -0,0 +1,186 @@
+defmodule Google.Protobuf do
+  @moduledoc """
+  Utility functions for working with Google Protobuf structs.
+  """
+
+  @doc """
+  Converts a `Google.Protobuf.Duration` struct to a value and `System.time_unit()`.
+
+  ## Examples
+
+      iex> to_time_unit(%Google.Protobuf.Duration{seconds: 10})
+      {10, :second}
+
+      iex> to_time_unit(%Google.Protobuf.Duration{seconds: 20, nanos: 100})
+      {20_000_000_100, :nanosecond}
+
+  """
+  @spec to_time_unit(Google.Protobuf.Duration.t()) :: {integer(), System.time_unit()}
```

Comment:
Heads up, now that we have https://hexdocs.pm/elixir/main/Duration.html, we should change these functions to match.

Reply:
We would need to bump elixir requirement, though

Reply:
I think it's fine to keep as it is?

Reply:
Maybe we can **accept** it in `from_time_unit` if using Elixir >= 1.17, but continue returning ` {integer(), System.time_unit()}` on `to_time_unit`

Reply:
If we don't return `Duration`, we're introducing a function now that is bound to be deprecated soon. I think I wouldn't define `to_time_unit/1` in the first place for now, or only defined it if compiled with Elixir 1.17+.

Reply:
Same for `from_time_unit`

Reply:
Removed them for now, we can consider adding them at some point :)